evalml/pipelines/utils.py

```diff
@@ -147,6 +148,55 @@ class TemplatedPipeline(_get_pipeline_base_class(problem_type)):
     return TemplatedPipeline({c.name: c.parameters for c in component_instances})
 
 
+def generate_pipeline_code(element):
+    """Creates and returns a string that contains the Python imports and code required for running the EvalML pipeline.
+
+    Arguments:
+        element (pipeline instance): The instance of the pipeline to generate string Python code for
+
+    Returns:
+        String representation of Python code that can be run separately in order to recreate the pipeline instance.
+        Does not include code for custom component implementation
+    """
+    component_names = [c.name for c in all_components()]
+    # hold the imports needed and add code to end
+    code_strings = []
+    if not isinstance(element, PipelineBase):
+        raise ValueError("Element must be a pipeline instance, received {}".format(type(element)))
+
+    component_graph = [com.name for com in element.component_graph]
+    custom_components = [(com.__class__.__name__, com.name) for com in element.component_graph if com.name not in component_names]
+    import_strings = [com.__class__.__name__ for com in element.component_graph if com.name in component_names]
```

Comment:
As with the component code, let's use `com.__class__` instead of `com.__class__.__name__` when we need to do these comparisons